domain/schema/controller.go

```diff
@@ -233,7 +237,43 @@ CREATE TABLE cloud_ca_cert (
 );
 
 CREATE UNIQUE INDEX idx_cloud_ca_cert_cloud_uuid_ca_cert
-ON cloud_ca_cert (cloud_uuid, ca_cert);`)
+ON cloud_ca_cert (cloud_uuid, ca_cert);
+
+CREATE TABLE cloud_credential (
+        uuid                TEXT PRIMARY KEY,
+        cloud_uuid          TEXT NOT NULL,
+        auth_type_id        TEXT NOT NULL,
+        owner_uuid          TEXT NOT NULL,
+        name                TEXT NOT NULL,
+        revoked             BOOLEAN,
+        invalid             BOOLEAN,
+        invalid_reason      TEXT,
+        CONSTRAINT chk_name_empty CHECK (name != ""),
+        CONSTRAINT          fk_cloud_credential_cloud
+            FOREIGN KEY         (cloud_uuid)
+            REFERENCES          cloud(uuid)
+        CONSTRAINT          fk_cloud_credential_auth_type
+            FOREIGN KEY         (auth_type_id)
+            REFERENCES          auth_type(id)
+--        CONSTRAINT          fk_cloud_credential_XXXX
```

Comment:
Do we need the comments?

Reply:
yes, this is a placeholder so remind us to add the FK when users are modelled. also used in model migration table